src/helpers/CamelJBang.ts

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import { globSync } from "glob";
+import { ShellExecution, ShellExecutionOptions, workspace, WorkspaceFolder } from "vscode";
+import { arePathsEqual, getCurrentWorkingDirectory } from "./helpers";
+
+export enum RouteOperation {
+	start = 'start',
+	stop = 'stop',
+	suspend = 'suspend',
+	resume = 'resume'
+}
+
+/**
+ * Camel JBang class which allows shell execution of different JBang CLI commands
+ */
+export class CamelJBang {
+
+	private camelJBangVersion: string;
+
+	constructor(private readonly jbang: string = 'jbang') {
+		this.camelJBangVersion = workspace.getConfiguration().get('kaoto.camelJBang.Version') as string;
+	}
+
+	public init(file: string): ShellExecution {
+		return new ShellExecution(this.jbang, [`'-Dcamel.jbang.version=${this.camelJBangVersion}'`, 'camel@apache/camel', 'init', `'${file}'`]);
+	}
+
+	// Camel Run with '--console' and specified port '--port=XXXX' is used for fetching data of running local integrations
+	// it is easier to fetch data and parse a JSON format instead of 'camel get route' option, where it is needed to parse shell stdout
+	public run(filePattern: string, cwd?: string, port?: number): ShellExecution {
+		const shellExecOptions: ShellExecutionOptions = {
+			cwd: cwd
+		};
+		return new ShellExecution(this.jbang,
+			[`'-Dcamel.jbang.version=${this.camelJBangVersion}'`,
+				'camel@apache/camel',
+				'run',
+				filePattern,
+				'--dev',
+				'--logging-level=info',
+				'--console',
+				`--port=${port ? port : 8080}`,
+				this.getCamelVersion(),
+				this.getRedHatMavenRepository(),
+				...this.getExtraLaunchParameter()
+			].filter(function (arg) { return arg; }), // remove ALL empty values ("", null, undefined and 0)
+			shellExecOptions
+		);
+	}
+
+	public stop(name: string): ShellExecution {
+		return new ShellExecution(this.jbang,
+			[`'-Dcamel.jbang.version=${this.camelJBangVersion}'`,
+				'camel@apache/camel',
+				'stop',
+				name // TODO when running using '*' then log JBang option is not working properly
```

Comment:
In which case it needs to be called with *?

Please add the related Camel jira for this issue (I suppose this is an upstream issue)

Reply:
> In which case it needs to be called with *?

or you mean that if previously there was a `camel run *`, then the camel stop is not workign properly?

Reply:
^ yes, that is a problem raised in a comment